commit 31727b0662fe8714509166fd322cd55013fa2a0c
Author: Valentin Raevsky <valentin@compulab.co.il>
Date:   Wed Jun 17 09:59:44 2015 +0300

    ARM: i.MX6: cm-fx6: Fix usdhc3 controller's property
    
    Fix usdhc3 controller's property in ordert to match
    the sbc-fx6, sbc-fx6m hw implementation.
    
    Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>

diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
index 9fed3d9..76df2e3 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
@@ -84,6 +84,7 @@
 
 &usdhc3 {
 	wp-gpios = <&gpio7 0 0>;
+	cd-gpios = <&gpio7 1 0>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
index c92136c..ec9bec3 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
@@ -94,6 +94,7 @@
 };
 
 &usdhc3 {
+	non-removable;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi
index 01f73ae..f98a87d 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi
@@ -103,8 +103,8 @@
 	pinctrl-0 = <&pinctrl_usdhc3>;
 	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
-	cd-gpios = <&gpio7 1 0>;
 	no-1-8-v;
+	keep-power-in-suspend;
 	vmmc-supply = <&reg_3p3v>;
 	status = "disabled";
 };
